Automattic/newspack-blocks
# feat(homepage-posts): infinite scroll

### All Submissions:






### Changes proposed in this Pull Request:



Closes https://github.com/Automattic/newspack-blocks/issues/748.

### How to test the changes in this Pull Request:

1. Go to a block with 'load more' button enabled
2. Enable Infinite Scroll on the block

<img width="266" alt="Screenshot 2024-08-22 at 12 36 51 PM" src="https://github.com/user-attachments/assets/9730a7e8-fb88-45a3-a83a-37e90ca9afd8">

3. See block on frontend and observe infinite scroll
4. Also try placing multiple instances of the Homepage Posts blocks with the same query attributes + infinite scroll enabled side by side on the same page (for example, in columns like this):

<img width="1348" alt="Screenshot 2024-09-10 at 2 47 49 PM" src="https://github.com/user-attachments/assets/98b66e9b-1227-4abd-a667-82990877cda1">

5. On the front-end, confirm that all instances of the block load more posts with infinite scroll, but without showing duplicate posts across all instances. Note that the order in which the block instances load more posts is uncontrolled and just depends on the order in which they fire their fetch requests.

See an example in production here: https://lebtown.com/

Scroll to the bottom of the page and see infinite scroll work.

### Other information:



> There is no e2e test framework setup in `newspack-blocks`. If one was added, I would add tests for this feature.

> n/a
